### PR TITLE
superwhisper 1.43.2 (new cask)

### PR DESCRIPTION
**Pull request by nihaals**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/s/superwhisper.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  auto_updates true
+  depends_on macos: ">= :ventura"
+
+  app "superwhisper.app"
+
+  uninstall quit: "com.superduper.superwhisper"
+
+  zap trash: [
+    "~/Documents/superwhisper",
```
Comment (bevanjkay) [inline Casks/s/superwhisper.rb:23]: Does this file store any dictations, if so we shouldn't trash it, it could be added as `rmdir` though.

Comment (nihaals) [reply Casks/s/superwhisper.rb:23]: `~/Documents/superwhisper` by default stores most of the app config as well as local recordings. I'm used to rmdir only really being used when a folder is shared amongst other apps and has a parent folder that can be deleted if all the other apps are also uninstalled and I feel like this use case isn't much different from other apps' app data. https://github.com/Homebrew/homebrew-cask/blob/1ee4f3ad9be282f1ccc6043200cac588198e5d97/Casks/r/rewind.rb#L30 also comes to mind which can store >100GB of screen recordings which would probably be even worse if deleted unknowingly.

Comment (bevanjkay) [reply Casks/s/superwhisper.rb:23]: To be clearer, in terms of `zap` we're primarily concerned with *not* deleting user-created data, but we should be deleting all preferences, caches etc...

I haven't used `superwhisper` so I'm happy to accept your judgement on what is best here.

Comment (nihaals) [reply Casks/s/superwhisper.rb:23]: If the goal is to not delete user-generated data then I think it makes sense to not include, although I've actually never noticed this before. For something like Microsoft Word this makes complete sense, I wouldn't expect the zap to delete all my documents created in a default location, although with this I think it's harder to draw the line.

With some apps I would consider preferences to actually be a big part of the app and you might spend significant time setting it up, and in the case of this folder, it includes what are essentially preferences the user has set up along with WAV files for local recordings. I think it's hard to pick a bucket of what type of data these recordings are though: I personally consider the history feature a part of the app (and the recordings just app data) and have never thought about using the recorded files outside of the app, but perhaps to someone else they use it similar to Voice Memos and use the WAV files directly.

I can provide any additional context about the app if needed but this question might be more about the goals of zap from here. As a user I'd expect zap to delete this folder, but I also expect zap to delete essentially everything (unless it is something like documents/projects), which might not actually be the goal.

Comment (bevanjkay) [reply Casks/s/superwhisper.rb:23]: From what you've described I'm okay with keeping this in the `zap`, and thank you for the thorough response.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**